packages/components/src/components/dropdown/dropdown.jsx

```diff
@@ -129,7 +130,7 @@ const Dropdown = ({
     }, [is_nativepicker, is_nativepicker_visible]);
 
     React.useEffect(() => {
```

Comment:
Would it be possible to use `useLayoutEffect` instead here? Then line 249 to 253 could be removed

Reply:
@oskar-binary If you mean just switching from `useEffect` to `useLayoutEffect` without using the `initial_render` ref, No! it doesn't work. I tried it before.
The issue is `useEffect` runs immediately when the `useEffect` dependency state (`is_list_visible`) is initialized. But we don't what this. we want the first execution of this effect happens after first real state change. (e.g. In **Vue.js** there is a `immediately` flag for controlling this behaviour but `useEffect` doesn't have such a flag)
A neater way is using a custom hook like this:

```js
const useDidMountEffect = (func, deps) => {
    const didMount = useRef(false);

    useEffect(() => {
        if (didMount.current) func();
        else didMount.current = true;
    }, deps);
}
```

But I thought we are not going to use this a lot and just implemented the behaviour locally.

Reply:
@mahdiaryayi-fs ah ok that makes sense, that hook looks nice 👍 but yeah not sure if we'll need this anywhere else.